reframe/core/schedulers/pbs.py

```diff
@@ -202,7 +209,12 @@ def poll(self, *jobs):
                      f'assuming all jobs completed')
             for job in jobs:
                 job._state = 'COMPLETED'
-                job._completed = True
+                if job.cancelled or out_ready(job):
+                    job._completed = True
+                else:
+                    self.log(f'Job {job.jobid} output has not been written '
+                             f'back to working directory yet; will poll '
+                             f'again later')
```

Comment:
No need for this imho.

Reply:
I think it's better if we leave a message. Before that we print in debug:
```
Return code is {completed.returncode}: assuming all jobs completed
```
So it can be misleading if we think all the jobs have completed.

Reply:
I don't like that fact that we have an `else` branch just to log something. First, I would change the misleading log message that you cite here; it's definitely not accurate. You can simply remove the `: assuming [...]` part. Second, instead of logging in the `else` branch, log in the main branch the opposite thing, e.g., `f'job {job.jobid} has finished'`.